lib/vscode/src/vs/platform/extensionManagement/node/extensionsScanner.ts

```diff
@@ -276,19 +276,20 @@ export class ExtensionsScanner extends Disposable {
 
 	private async scanExtensionsInDir(dir: string, type: ExtensionType): Promise<ILocalExtension[]> {
 		const limiter = new Limiter<any>(10);
-<<<<<<< HEAD
-		const extensionsFolders = await pfs.readdir(dir)
+		/*
+		 * NOTE@coder: use fileService.resolve() like upstream does,
+		 * but simply ignore directories that do not exist. (upstream does not)
+		 * 
+		 * Used to (<1.54) use pfs.readdir.
+		 */
+		const stat = await this.fileService.resolve(URI.file(dir))
```

Comment:
I think we could save this for a separate PR - not sure if we include any extension paths that may not be writable by the user; in that case, creating the directory would fail, and then so would this.

Reply:
Ahh good point. I'm partially inclined to let the failure happen since it seems like a user error that should be surfaced. Definitely down to visit this in another PR though.

Reply:
Cool! Conclusion settled: @oxy to open follow-up PR